.tekton/scripts/check-partner-tasks.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$SCRIPTDIR/../.." || exit 1
+
+# Expected partner task directory structure:
+# task/partners/<task name>/<version>/<task file>
+
+if [ ! -e "partners/" ]; then
+  exit 0
+fi
+
+check_dir_structure() {
+  issue_found=0
+  for task_dir in partners/*; do
+    task_name=$(basename "$task_dir")
+    for version_dir in "$task_dir"/*; do
+      task_version=$(basename "$version_dir")
+      if ! echo "$task_version" | grep "^[0-9.]\+$" >/dev/null 2>&1; then
+        echo "error: $version_dir is not a version directory. A version directory name only contains digits and dot, e.g. 0.1, 0.2.1"
+        issue_found=1
+      else
+        # Ignore empty directory whatever it is a version directory or not.
+        if [ -n "$(ls "$version_dir")" ]; then
+          found=
+          for yaml_file in "$version_dir"/*.yaml; do
+            task_file=$(basename "$yaml_file")
+            if [ "${task_name}.yaml" == "$task_file" ]; then
+              found=1
+              break
+            fi
+          done
+          if [ -z "$found" ]; then
+            echo "error: no task file is found under $version_dir. A task file must be named with the same task name. For example, task name is task1, then task file must have name task1.yaml"
+            issue_found=1
+          fi
+        fi
+      fi
+    done
+  done
+  return $issue_found
+}
+
+check_privilege_use() {
+  resultf=$(mktemp)
+
+  find partners/*/*/*.yaml | awk -F '/' '{ print $0, $2, $4 }' | \
+  while read -r task_file task_name yaml_file
+  do
+    if [ "${task_name}.yaml" != "$yaml_file" ]
+    then
+      continue
+    fi
+    while IFS=: read -r sc_config check_value err_msg
+    do
+      task_idx=0
+      # When no securityContext config is set, yq outputs message "Error: no matches found" to stderr.
+      # Redirect this message into stdout in order to suppress these potential Errors, and such error
+      # messages does not match any config value obviously.
+      for value in $(yq -e ".spec.steps[].securityContext.$sc_config" "$task_file" 2>&1)
+      do
+        if [ "$value" == "$check_value" ]
+        then
+          echo "found" >"$resultf"
+          step_name=$(yq -e ".spec.steps[$task_idx].name" "$task_file")
+          echo "error: in step $step_name, $err_msg Task $task_file"
+        fi
+        task_idx=$((task_idx+1))
+      done
+    done <<EOF
+allowPrivilegeEscalation:true:allowPrivilegeEscalation is not allowed to be true to request more privileges.
+runAsUser:0:root(0) is not allowed to be set to securityContext.runAsUser.
+privileged:true:ecurityContext.privileged is not allowed to be true.
+EOF
+  done
+  rv=0
+  if [ -n "$(cat "$resultf")" ]
+  then
+    rv=1
+  fi
+  return $rv
+}
+
+check_dir_structure || exit 1
+check_privilege_use || exit 1
```

Comment:
when check_dir_structure will fail then check_privilege_use will not be executed

Reply:
Yes, this is intended. An expected directory structure is a prerequisite so that `check_privilege_use` can find out the task files.